easybuild/tools/run.py

```diff
@@ -112,31 +114,37 @@ def print(self):
         Report failed shell command for this RunShellCmdError instance
         """
 
-        def pad_4_spaces(msg):
-            return ' ' * 4 + msg
+        def pad_4_spaces(msg, color=None):
+            padded_msg = ' ' * 4 + msg
+            if color:
+                return colorize(padded_msg, color)
+            else:
+                return padded_msg
+
+        caller_file_name, caller_line_nr, caller_function_name = self.caller_info
+        called_from_info = f"'{caller_function_name}' function in {caller_file_name} (line {caller_line_nr})"
 
         error_info = [
             '',
-            "ERROR: Shell command failed!",
+            colorize("ERROR: Shell command failed!", COLOR_RED),
             pad_4_spaces(f"full command              ->  {self.cmd}"),
             pad_4_spaces(f"exit code                 ->  {self.exit_code}"),
+            pad_4_spaces(f"called from               ->  {called_from_info}"),
             pad_4_spaces(f"working directory         ->  {self.work_dir}"),
         ]
 
         if self.out_file is not None:
             # if there's no separate file for error/warnings, then out_file includes both stdout + stderr
             out_info_msg = "output (stdout + stderr)" if self.err_file is None else "output (stdout)         "
-            error_info.append(pad_4_spaces(f"{out_info_msg}  ->  {self.out_file}"))
+            error_info.append(pad_4_spaces(f"{out_info_msg}  ->  {self.out_file}", color=COLOR_YELLOW))
```

Comment:
Hmm, the yellow here is maybe not a good idea when terminal doesn't have a dark background, should be changed to different color that works on both dark/light background?

Reply:
Yeah, it might be safer to use blue. Although any sane color theme will have a yellow with enough contrast with the light background.

Reply:
Looks OK for me with yellow on a white background, and with (standard) blue it looks horrible on dark background, so sticking to yellow...

![Screenshot 2024-10-14 at 09 47 14](https://github.com/user-attachments/assets/90b8f3d2-a13d-4ffb-90a8-7c518e2c2751)